### ISSUE TITLE
no failed queue table > job not deleted > infinite loop

### ISSUE THREAD

**Issue by ctadlock** [laravel/framework#10678]
If a failed queue table doesn't exist in the database the `Queue/Worker` code will throw an exception when trying to write to it which is not caught, resulting in the job never being deleted causing an infinite loop of job processing; irrespective of `tries`. 

The methods `process` and `logFailedJobs` need to be totally reworked. My suggestion is the job deletion code should be moved up to the parent method `process`. There also needs to be a way to not have a failed queue table at all and optional leverage the failed job framework built into AWS SQS (when that is used).

![image](https://cloud.githubusercontent.com/assets/1539397/10616662/12e4a904-771b-11e5-9c3e-ae802a2bd4bb.png)

**Comment by GrahamCampbell** [comment]
We're open to PRs.

**Comment by ctadlock** [comment]
Closing an obvious bug. This is why Laravel is starting to get a bad reputation.

**Comment by GrahamCampbell** [comment]
You opened an issue and described how you'd fix it. Please PR that.

**Comment by ctadlock** [comment]
I gave contextual guidance on a proposed resolution given my specific understanding of the issue, but limited understanding of the implementation of the framework as a whole. Part of this is not a simple code change. The framework apparently is assuming the presence of `failure`; which is set somewhere else in the framework with potentially other dependencies. I could bumble my way through the change and unknowingly break something else. It all comes down to efficiency and quality. The maintainers of the framework are in the best position to make more significant changes. As a user of the framework I identified the issue and gave you enough information to be able to both quickly understand and resolve the issue. It is an unreasonable expectation, and very inefficient, to expect all users to become framework implementation experts. Lets not forget, in my opinion this is a very high priority bug as it has the potential to crash the server and is easy to make happen. Nowhere is the documentation does it state "You must have a failed jobs table otherwise when a job throws an exception it will cause an infinite loop.". I discovered this issue when my production server crashed.  

Ill put in a PR for part of this, but supporting not having a failed jobs table seems best left to the maintainers.
